daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -308,8 +308,8 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
   final def fold[A](z: A)(f: (A, (Nid, Node.GenNode[Nid, Cid])) => A): A = {
     var acc = z
     foreach { (nodeId, node) =>
-      // make sure to not tie the knot by mistake by evaluating early
-      val acc2 = acc
+      // make sure to not tie the knot by mistake by evaluating early //TODO: ???
+      val acc2 = acc //TODO: why is this necessary?
       acc = f(acc2, (nodeId, node))
```

Comment:
@stephencompall-DA could you comment on that ?

Reply:
If `f` was lazy, then `acc = f(acc, ...)` is circular. Whether you wish to keep defending against that (I assume copy-paste or signature change) is your choice; it's harmless, anyhow.

Reply:
By the way, there actually was such a bug, fixed in this way. I don't remember where, though.